flagsmith-jira-app/src/frontend/components/ProjectSettingsPage.tsx

```diff
@@ -90,68 +96,79 @@ const ProjectSettingsForm = ({ setError, saveProjectId, ...props }: ProjectSetti
       console.error(error);
       return [];
     }
-  }, [props.projectId]);
+  }, [savedProjectIds]);
 
   if (projects === undefined) {
     return <Spinner label="Loading projects" />;
   }
 
   const onSave = async () => {
-    await saveProjectId(projectId ?? "");
+    await saveProjectIds(projectIds);
   };
 
   const onClear = async () => {
     // clear saved state
-    await saveProjectId("");
+    await saveProjectIds([]);
     // clear unsaved state
-    setProjectId("");
+    setProjectIds([]);
   };
 
-  const connected = !!currentProject && features && features.length > 0;
+  const connected = !!currentProjects && features && features.length > 0;
 
   return (
     <Fragment>
       <Box xcss={{ marginBottom: "space.300" }}>
         <Inline space="space.050" alignBlock="center">
-          <Strong>Project:</Strong> {!!currentProject && <Text>{currentProject.name}</Text>}
-          {!currentProject && projects && projects.length > 0 && !connected && (
-            <Lozenge appearance="moved">Not connected</Lozenge>
+          <Strong>Project(s):</Strong>
+          {(currentProjects?.length ?? 0) > 0 && (
+            <Text>{currentProjects!.map((project) => project.name).join(", ")}</Text>
           )}
-          {projects && projects.length === 0 && (
-            <Lozenge appearance="removed">No projects to connect</Lozenge>
+          {currentProjects?.length === 0 && projects?.length > 0 && !connected && !hasChanges && (
+            <Lozenge appearance="moved">Not connected</Lozenge>
           )}
-          {!!currentProject && !!features && !connected && (
+          {projects?.length === 0 && <Lozenge appearance="removed">No projects to connect</Lozenge>}
+          {(currentProjects ?? []).length > 0 && !!features && !connected && !hasChanges && (
             <Lozenge appearance="removed">No features to connect</Lozenge>
           )}
-          {connected && <Lozenge appearance="success">Connected</Lozenge>}
+          {connected && !hasChanges && <Lozenge appearance="success">Connected</Lozenge>}
```

Comment:
As I recall I omitted `hasChanges` from some of these to reduce flickering at the cost of the status being out of date for a moment or two, but this is all quite tricky so if this works I'd not change it again 😀